frontend/src/hooks/useAuth.tsx

```diff
@@ -60,7 +69,11 @@ const useProvideAuth = (): AuthContextType => {
     .then(res => {
       setUser(res.data.user)
     })
-  }, [])
+    .catch(err => {
+      logout();
+      history.replace('/login'); // not logged in so redirect to login.
+    })
+  }, [history])
```

Comment:
This seems redundant (doing the same thing as the `<Redirect/>` components in `RedirectedRoute`). Can one of them be removed?

Reply:
If it works without the history replace line, if we keep it in should render the login page faster anyway, so I think it is good.

Reply:
oh i see. The other case doesnt check the jwt. This case is where if the jwt is not validated by the backend then we need to logout and go to `/login`